payments.py

```diff
@@ -0,0 +1,40 @@
+from django.contrib.auth.models import Permission, User
+
+from memberships.models import Member, Membership, FailedPayment, Payment
+from funky_time import epoch_to_datetime
+
+
+def handle_stripe_payment(event):
+    if event["type"] == "invoice.payment_failed":
+        FailedPayment.objects.create(
+            stripe_customer_id=event["data"]["object"]["customer"],
+            stripe_subscription_id=event["data"]["object"]["subscription"],
+            stripe_event_type=event["type"],
+        )
+    if event["type"] == "invoice.payment_succeeded":
+        member = Member.objects.get(email=event["data"]["object"]["customer_email"])
+        log_payment(event, member)
+        update_last_payment(event, member)
+        add_user_sand_permission(member)
+
+
+def log_payment(event, member):
+    # Log payment for a member in the database
+    Payment.objects.create(
+        member=member,
+        stripe_subscription_id=event["data"]["object"]["subscription"],
+    )
+
+
+def update_last_payment(event, member):
+    # Store payment DateTime in membership model
+    membership = Membership.objects.get(member=member)
+    membership.last_payment_time = epoch_to_datetime(event["created"])
+    membership.save()
```

Comment:
I believe you should be able to get the membership from the `member` object directly, without querying the db.

```suggestion
    membership = member.membership
    membership.last_payment_time = epoch_to_datetime(event["created"])
    membership.save()
```

Although double check before you do this because I'm not 100% certain :)

Reply:
`AttributeError: 'Member' object has no attribute 'membership'`

This makes sense to me as the foreign key is in `membership` as opposed to `member`